File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.py
class Solution:
    def closetTarget(self, words: List[str], target: str, startIndex: int) -> int:
        n = len(words)
        moves = 0
        while(moves * 2 <= n):
            if(words[(moves + startIndex)%n] == target or words[(startIndex - moves + n)%n] == target):
                return moves
            moves += 1
        return -1

File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.cpp
class Solution {
public:
    int closetTarget(vector<string>& words, string target, int startIndex) {
        int sizeW = words.size();
        int moves;
        for(moves = 0;  moves*2 <= sizeW ; moves++){
            if(words[(startIndex + moves)%sizeW] == target) return moves;
            if(words[(startIndex - moves + sizeW)%sizeW] == target) return moves;
        }
        return -1;
    }
};